common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesBrokerApplicationMetadataCache.java

```diff
@@ -78,7 +96,14 @@ public void remove(@NonNull final String clientId,
     private void disposeOfDuplicateRecords(@NonNull final String clientId,
                                            @NonNull final String environment,
                                            final int uid) {
-        final List<BrokerApplicationMetadata> allMetadata = getAll();
+        Set<BrokerApplicationMetadata> allMetadata = null;
+        readLock.lock();
```

Comment:
You'll need to hold the write lock for the entire duration of this activity. Read locks aren't exclusive.